internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -664,7 +674,31 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *ai
 
 		return nil
 	}
-	return deployment, mutateFn
+	return deployment, mutateFn, nil
+}
+
+func getOpenshiftVersions(instance *aiv1beta1.AgentServiceConfig) (string, error) {
+	if instance.Spec.OSImages == nil {
+		return OpenshiftVersions(), nil
+	}
+
+	var versions models.OpenshiftVersions
+	for _, image := range instance.Spec.OSImages {
+		version := models.OpenshiftVersion{
```

Comment:
What do you say about setting the `DisplayName` to the value of `Major.Minor`? I think assisted-service does this already but it's nicer if we send the configs already properly formated.